### PR TITLE
add country argument to organisations, add tests, improve documentation

### PR DESCRIPTION
**Pull request by PietrH**
### Add country argument

I've added country argument to `rgbif::organizations()`, similar as it's schedueled to be implemented in `rgbif::dataset_search()`, because the parameter still has the "YET TO BE IMPLEMENTED" prefix in dataset.r I couldn't use the r-oxygen template. But I used the same description and example, I did add quotation marks to emphasize character input. 

### Added tests 

- Added test to see if the requested country was actually returned
- Added test to see if at least one GBIF Secretariat dataset can be found (eg. the taxonomic backbone)
- Added test to see if returned dataframe has any duplicated rows 
- Added test to check for error messages when a bad uuid or datatype is passed

No tests for errors on `start` or `offset`, start will return errors for invalid values (strings, Inf, ...), offset will not, in both cases the function will still return an object (and use defaults). 

### Improvements to documentation

Documentation hadn't been updated since the last changes, so `devtools::document()` resulted in a change to `lit_search.Rd`

#### Value
Currently:
> A list of length one or two. If uuid is NULL, then a data.frame with call metadata, and a data.frame, but if uuid given, then a list.

Is unclear to me, also doesn't seem to correspond with current behavior:

``` r
library(rgbif)
# uuid is NULL
organizations() |> dplyr::glimpse()
#> List of 2
#>  $ meta:'data.frame':    1 obs. of  4 variables:
#>  $ data: tibble [100 × 31] (S3: tbl_df/tbl/data.frame)

# uuid is given
organizations(uuid = "fbca90e3-8aed-48b1-84e3-369afbd000ce") |> dplyr::glimpse()
#> List of 2
#>  $ meta: NULL
#>  $ data:List of 24

# uuid is given, and data is specified
organizations(uuid = "fbca90e3-8aed-48b1-84e3-369afbd000ce",
              data = "contact") |> dplyr::glimpse()
#> List of 2
#>  $ meta: NULL
#>  $ data: tibble [2 × 15] (S3: tbl_df/tbl/data.frame)
```

Changed to:

> A list of length of two, consisting of a data.frame \code{$meta} when uuid is NULL, and \code{$data} which can either be a list or a data.frame depending on the requested type of data

#### Arguments

Currently:
> (character) UUID of the data node provider. This must be specified if data is anything other than 'all'.

This is not true, and the package even returns an error confirming it's not true: 

```
Error in getdata(data) : 
  You must specify a uuid if data does not equal "all" and
       data does not equal of deleted, pending, or nonPublishing
```

Changed to:

> (character) UUID of the data node provider. This must be specified if data is anything other than 'all', 'deleted', 'pending', or 'nonPublishing'.

I tried to stick to the style of the existing documentation of the function. Any further clarifications welcome!

## Related Issue
#606 

## Example
```r 
rgbif::organizations(country = "ANDORRA")

```

### CODE REVIEW

File: R/organizations.r [hunk 1]
```diff
@@ -8,19 +8,23 @@
...
+#' @return A list of length of two, consisting of a data.frame \code{meta} when
+#'  uuid is NULL, and \code{data} which can either be a list or a data.frame
+#'  depending on the requested type of data.
 #'
 #' @references <https://www.gbif.org/developer/registry#organizations>
 #'
 #' @examples \dontrun{
 #' organizations(limit=5)
 #' organizations(query="france", limit=5)
+#' organizations(country = "SPAIN")
```
Comment (jhnwllr) [inline R/organizations.r:27]: This has to be a country code I believe. 
ISO 639-1 (2 letter) country code.

**Comment by jhnwllr** [comment]
> 35 tests fail, I'll include a log. As far as I can see, the main branch is currently failing, I'll include a log.
> 
> [test-all.Rout.txt](https://github.com/ropensci/rgbif/files/10723501/test-all.Rout.txt)

These vcr errors usually aren't important. You might need to delete some yml fixtures.

File: R/organizations.r (hunk 1 above)
Comment (PietrH) [reply R/organizations.r:27]: Correct, this should also be changed in the r-oxygen template `dataset.r`, it does work most of the time, but not for "SAINT_BARTHELEMY". 

Do you prefer it if users enter isocodes$gbif_name (ANTIGUA_BARBUDA) or isocodes$code (AG)? We can do it either way and still pass the correct argument to the API.

Comment (jhnwllr) [reply R/organizations.r:27]: I think it's better to use the 2 letter isocode only. This is what is done throughout the rest of rgbif, so I think it should be consistent.

Comment (PietrH) [reply R/organizations.r:27]: Alright, I've changed it in [4ad8ec4](https://github.com/ropensci/rgbif/pull/615/commits/4ad8ec413e255847a9a46efc95f8df48b674897e)

**Comment by PietrH** [comment]
Is it possible tests are failing because a github secret isn't being forwarded to the tests? 

in testthat.Rout (github actions workflow) errors seem to all be connected with missing credentials: 
> Actual message: "supply a username"

``` yml

    env:
      R_REMOTES_NO_ERRORS_FROM_WARNINGS: true
      CRAN: ${{ matrix.config.cran }}
      VCR_VERBOSE_ERRORS: true
      GEONAMES_USER: ${{ secrets.GEONAMES_USER }}
      GBIF_EMAIL: ${{ secrets.GBIF_EMAIL }}
      GBIF_USER: ${{ secrets.GBIF_USER }}
      GBIF_PWD: ${{ secrets.GBIF_PWD }}
```

**Comment by PietrH** [comment]
@jhnwllr , What is the status of your review? Is there anything I can change in the meantime?

**Comment by jhnwllr** [comment]
> @jhnwllr , What is the status of your review? Is there anything I can change in the meantime?

@PietrH sorry been sick for a few days. I will try to look at it today. First impression is that it's fine.

**Comment by jhnwllr** [comment]
> Is it possible tests are failing because a github secret isn't being forwarded to the tests?
> 
> in testthat.Rout (github actions workflow) errors seem to all be connected with missing credentials:
> 
> > Actual message: "supply a username"
> 
> ```yaml
>     env:
>       R_REMOTES_NO_ERRORS_FROM_WARNINGS: true
>       CRAN: ${{ matrix.config.cran }}
>       VCR_VERBOSE_ERRORS: true
>       GEONAMES_USER: ${{ secrets.GEONAMES_USER }}
>       GBIF_EMAIL: ${{ secrets.GBIF_EMAIL }}
>       GBIF_USER: ${{ secrets.GBIF_USER }}
>       GBIF_PWD: ${{ secrets.GBIF_PWD }}
> ```
> 
> The error stems from the helper:
> 
> ```r
> check_user <- function(x) {
>   z <- if (is.null(x)) Sys.getenv("GBIF_USER", "") else x
>   if (z == "") getOption("gbif_user", stop("supply a username")) else z
> }
> ```

Yes this is the reason these tests are failing. I am not sure how to fix that right now.

**Comment by PietrH** [comment]
> > @jhnwllr , What is the status of your review? Is there anything I can change in the meantime?
> 
> @PietrH sorry been sick for a few days. I will try to look at it today. First impression is that it's fine.

I hope you feel better soon, there is no rush. Just let me know if you require any changes.

File: tests/testthat/test-organizations.r [hunk 2]
```diff
@@ -27,3 +29,47 @@ test_that("returns the correct class", {
...
+    {
+      gd <- organizations(
+        data = "hostedDataset",
+        uuid = "fbca90e3-8aed-48b1-84e3-369afbd000ce"
+      )
+    },
+    preserve_exact_body_bytes = TRUE
+  )
+  # The GBIF Secretariat hosts at least one dataset
+  expect_gt(NROW(gd), 1)
```
Comment (jhnwllr) [inline tests/testthat/test-organizations.r:60]: I prefer lowercase nrow.

**Review by jhnwllr — CHANGES_REQUESTED** [review]
looks good. Thanks!

**Pull request merged**